File: src/test/java/no/nav/aareg/teknisk/historikk/ApiDocsTest.java
package no.nav.aareg.teknisk.historikk;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.web.servlet.client.RestTestClient;

@DirtiesContext
class ApiDocsTest extends AaregTekniskHistorikkTest {

    @Autowired
    private RestTestClient restTestClient;

    @Test
    void apiDocsEndepunktErTilgjengelig() {
        restTestClient.get().uri("/api-docs").exchange()
                .expectStatus().isOk();
    }

    @Test
    void swaggerEndepunktErTilgjengelig() {
        restTestClient.get().uri("/swagger-ui/index.html").exchange()
                .expectStatus().isOk();
    }

    @Test
    void specFilErTilgjengelig() {
        restTestClient.get().uri("/openapi-spec.yaml").exchange()
                .expectStatus().isOk();
    }
}

File: src/test/kotlin/no/nav/aareg/teknisk_historikk/ApiDocsTest.kt
package no.nav.aareg.teknisk_historikk

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.http.HttpStatus
import org.springframework.test.annotation.DirtiesContext

@DirtiesContext
class ApiDocsTest : AaregTekniskHistorikkTest() {
    @Autowired
    lateinit var testRestTemplate: TestRestTemplate

    @Test
    fun `api-docs-endepunkt er tilgjengelig`() {
        val resultat = testRestTemplate.getForEntity("/api-docs", String::class.java)

        Assertions.assertEquals(HttpStatus.OK, resultat.statusCode)
    }

    @Test
    fun `swagger-endepunkt er tilgjengelig`() {
        val resultat = testRestTemplate.getForEntity("/swagger-ui/index.html", String::class.java)

        Assertions.assertEquals(HttpStatus.OK, resultat.statusCode)
    }

    @Test
    fun `spec-fil er tilgjengelig`() {
        val resultat = testRestTemplate.getForEntity("/openapi-spec.yaml", String::class.java)

        Assertions.assertEquals(HttpStatus.OK, resultat.statusCode)
    }
}